packages/@aws-cdk/aws-route53/lib/records/basic.ts

```diff
@@ -0,0 +1,365 @@
+  /**
+   * The hosted zone in which to define the new record.
+   */
+  readonly zone: IHostedZone;
+
+  /**
+   * The domain name for this record.
+   *
+   * @default zone root
+   */
+  readonly recordName?: string;
+
+  /**
+   * The resource record cache time to live (TTL) in seconds.
+   *
+   * @default 1800 seconds
+   */
+  readonly ttl?: number;
+}
+
+/**
+ * Construction properties for a BasicRecord.
+ */
+export interface BasicRecordProps extends RecordProps {
+  /**
+   * The record type.
+   */
+  readonly recordType: RecordType;
+
+  /**
+   * The values for this record.
+   */
+  readonly recordValues: string[];
+}
+
+/**
+ * A basic record
+ *
+ * @resource AWS::Route53::RecordSet
+ *
+ * @see https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-values-basic.html
+ */
+export class BasicRecord extends Resource {
+  constructor(scope: Construct, id: string, props: BasicRecordProps) {
+    super(scope, id);
+
+    const ttl = props.ttl === undefined ? 1800 : props.ttl;
+
+    new CfnRecordSet(this, 'Resource', {
+      hostedZoneId: props.zone.hostedZoneId,
+      name: determineFullyQualifiedDomainName(props.recordName || props.zone.zoneName, props.zone),
+      type: props.recordType,
+      resourceRecords: props.recordValues,
+      ttl: ttl.toString()
+    });
+  }
+}
+
+/**
+ * Construction properties for a ARecord.
+ */
+export interface ARecordProps extends RecordProps {
+  /**
+   * The IP addresses.
+   */
+  readonly ipAddresses: string[];
+}
+
+/**
+ * A DNS A record
+ *
+ * @resource AWS::Route53::RecordSet
+ */
+export class ARecord extends BasicRecord {
+  constructor(scope: Construct, id: string, props: ARecordProps) {
+    super(scope, id, {
+      ...props,
+      recordType: RecordType.A,
+      recordValues: props.ipAddresses,
+    });
+  }
+}
+
+/**
+ * Construction properties for a AaaaRecord.
+ */
+export interface AaaaRecordProps extends RecordProps, ARecordProps {}
+
+/**
+ * A DNS AAAA record
+ *
+ * @resource AWS::Route53::RecordSet
+ */
+export class AaaaRecord extends BasicRecord {
+  constructor(scope: Construct, id: string, props: AaaaRecordProps) {
+    super(scope, id, {
+      ...props,
+      recordType: RecordType.AAAA,
+      recordValues: props.ipAddresses,
+    });
+  }
+}
+
+/**
+ * Construction properties for a CnameRecord.
+ */
+export interface CnameRecordProps extends RecordProps {
+  /**
+   * The domain name.
+   */
+  readonly domainName: string;
+}
+
+/**
+ * A DNS CNAME record
+ *
+ * @resource AWS::Route53::RecordSet
+ */
+export class CnameRecord extends BasicRecord {
+  constructor(scope: Construct, id: string, props: CnameRecordProps) {
+    super(scope, id, {
+      ...props,
+      recordType: RecordType.CNAME,
+      recordValues: [props.domainName]
+    });
+  }
+}
+
+/**
+ * Construction properties for a TxtRecord.
+ */
+export interface TxtRecordProps extends RecordProps {
+  /**
+   * The text values.
+   */
+  readonly values: string[];
```

Comment:
This property class will have both `recordValues` and `values`. How am I supposed to interpret that as a user?

Reply:
No, it won't because I'm extending `RecordProps` here and not `BasicRecordProps`

Reply:
Got it.